news/support_gather_parallel.rst

```diff
@@ -0,0 +1,23 @@
+**Added:**
+
+* ``openfe gather`` now supports replicates that have been submitted in parallel across separate directories.
```

Comment:
@IAlibay will this be clear to users, or is there some better language I could be using?

Reply:
I think we could make this very clear by having something similar to [this](https://github.com/OpenFreeEnergy/gufe/blob/339143f86ffe234e474ac3313a9fb3ddb90fca98/gufe/chemicalsystem.py#L14) from the industry benchmarking but in the main CLI docs and point to that, there we could add more details on the expected file structure?

Reply:
I agree- opened an issue [here](https://github.com/OpenFreeEnergy/openfe/issues/1063) since this is outside of the scope of this PR.